File: readability.py
import cs50

#setting everything for the beginning
letter_count = 0
word_count = 1
sentence_count = 0

#user input
text = cs50.get_string("The Text: ")
text_length = len(text)

#counting letters
for i in range(text_length):
    if(text[i].isalpha()):
        letter_count += 1

#counting words
    if(text[i].isspace()):
        word_count += 1

#counting sentences
    if(text[i] == "." or text[i] == "?" or text[i] == "!"):
        sentence_count += 1

#calculating
index = (0.0588 * letter_count / word_count * 100) - (0.296 * sentence_count / word_count * 100) - 15.8

finality = round(index)

if finality < 1:
    print("Before grade 1 ")

elif finality > 16:
    print("Grade 16+ ")

else:
    print(f"Grade {finality}")

File: readability.c
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int main(void)
{
    // Ask for input
    string text = get_string("The text:\n ");
    
    // Defining the variables
    int let, sen, wor;
    
    int sl = strlen(text);
    
    let = 0;
    sen = 0;
    wor = 0;
    
    // Getting into the loop
    for(int z = 0; z < sl; z++)
    {
        //for each character calculate
        char i = text[z];
        
        if (isalpha(i) != 0)
        {
            let++;
        }
    
        if (i == '.' || i == '?' || i == '!')
        {
            sen++;
        }
        
        if (i == ' ')
        {
            wor++;
        }
    }
// Otherwise it is not gonna count the last word
wor = wor + 1;
        
// Convert integers to floats and do the math
float L = ((float)let / (float)wor) * 100;
float S = ((float)sen / (float)wor) * 100;
        
    float index = 0.0588 * L - 0.296 * S - 15.8;
        
// Truncating
int finality = round(index);
        
    if (finality > 16)
    {
        printf("Grade 16+\n");
    }
        
    else if (finality < 1)
    {
        printf("Before grade 1\n");
    }

    else 
    {
        printf("Grade %i\n", finality);
    }
       
}